pkg/kptfile/kptfileutil/util.go

```diff
@@ -212,104 +196,33 @@ func UpdateKptfile(localPath, updatedPath, originPath string, updateUpstream boo
 	return nil
 }
 
-func UpdateUpstreamLock(path string, upstreamLock *kptfilev1.UpstreamLock) error {
-	const op errors.Op = "kptfileutil.UpdateUpstreamLock"
+// UpdateUpstreamLockFromGit updates the upstreamLock of the package specified
+// by path by using the values from spec. It will also populate the commit
+// field in upstreamLock using the latest commit of the git repo given
+// by spec.
+func UpdateUpstreamLockFromGit(path string, spec *git.RepoSpec) error {
+	const op errors.Op = "kptfileutil.UpdateUpstreamLockFromGit"
 	// read KptFile cloned with the package if it exists
-	kptfile, err := pkg.ReadKptfile(filesys.FileSystemOrOnDisk{}, path)
+	kpgfile, err := pkg.ReadKptfile(filesys.FileSystemOrOnDisk{}, path)
```

Comment:
spin off --- keeping this PR a pure revert is a good idea, but there's a minor local variable typo here that would be nice to fix in main

Reply:
Yeah :) I actually fixed this typo and much of my battling with rebases last night was trying to keep this typo fixed; ultimately I gave up; will fix in main.